### PR TITLE
[chore] Changing linter to ruff

### PR DESCRIPTION
**Pull request by bushwhackr**
## Description

+ Deprecating pylint, black, flake8 for ruff
  + We are still using black for linting generated sdk code
+ Using ruff as the new linter
+ Migrating linting steps to new Taskfile.lint.yaml
  + `task lint` remains the same
  + `task format` => `task lint:format`

```
# Before
task lint -f  749.99s user 19.68s system 519% cpu 2:28.25 total
# After
task lint -f  389.40s user 7.44s system 444% cpu 1:29.32 total
```

+ Updated vuln dependencies
  + certifi
  + tornado
  + urllib3

+ Documentation Changes are the result of formatting done by `ruff` against the docstrings

## Type of Change

Label this pull request to place it under the correct category in Release Notes:

|        **Pull Request Label**         | **Category in Release Notes** |
|:-------------------------------------:|:-----------------------------:|
|       `enhancement`, `feature`        |          🚀 Features          |
| `bug`, `refactoring`, `bugfix`, `fix` |    🔧 Fixes & Refactoring     |
|       `build`, `ci`, `testing`        |    📦 Build System & CI/CD    |
|              `breaking`               |      💥 Breaking Changes      |
|            `documentation`            |       📝 Documentation        |
|            `dependencies`             |    ⬆️ Dependencies updates    |

## Checklist

- [ ] I have read the [`CODE_OF_CONDUCT.md`](https://github.com/featurebyte/featurebyte/blob/main/CODE_OF_CONDUCT.md) and [`CONTRIBUTING.md`](https://github.com/featurebyte/featurebyte/blob/main/CONTRIBUTING.md) guides.
- [ ] I have written tests for the changes made.
- [ ] I have written docstrings in [NumpyDoc format](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)
- [ ] I have labeled my Pull Request correctly

### CODE REVIEW

File: featurebyte/__init__.py [hunk 1]
```diff
@@ -1,10 +1,12 @@
 """Python Library for FeatureOps"""
 
-from typing import Any, List, Optional
+# Ignore import errors
+# ruff: noqa: F401
```
Comment (bushwhackr) [inline featurebyte/__init__.py:5]: We have imports that are not used within the __init__.py. 
If we do not ignore F401, it will cause those import to be added to __all__

File: pyproject.toml [hunk 2]
```diff
@@ -260,3 +224,31 @@ markers = [
...
+quote-style = "double"
+indent-style = "space"
+skip-magic-trailing-comma = false
+line-ending = "auto"
+docstring-code-format = true
+docstring-code-line-length = "dynamic"
+
+[tool.ruff.lint]
+select = ["E1", "E4", "E7", "E9", "F", "I"]  # Rules to use by default. Consider adding "UP" for upgradability checks.
+ignore = []
```
Comment (bushwhackr) [inline pyproject.toml:245]: I did not add the pylint rules back yet. Will do another PR. Refer to: https://docs.astral.sh/ruff/rules/#pylint-pl

**Review by gee-senbong — APPROVED** [review]
